heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -379,6 +393,102 @@ export class JestPlugin implements IHeftPlugin<IJestPluginOptions> {
+          // Example:  <rootDir>/path/to/file.js
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(ROOTDIR_TOKEN.length));
+          propertyValue = path.resolve(options.buildFolder, restOfPath);
+        } else if (propertyValue.startsWith(CONFIGDIR_TOKEN)) {
+          // Example:  <configDir>/path/to/file.js
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(CONFIGDIR_TOKEN.length));
+          propertyValue = path.resolve(configurationFileDir, restOfPath);
+        } else if ((packageDirMatches = PACKAGEDIR_REGEX.exec(propertyValue)) !== null) {
+          // Example:  <packageDir:@my/package>/path/to/file.js
+          const packageName: string | undefined = packageDirMatches.groups?.[PACKAGE_CAPTUREGROUP];
+          if (!packageName) {
+            throw new Error(
+              `Could not parse package name from "packageDir" token ` +
+                (propertyName ? `of property "${propertyName}" ` : '') +
+                `in "${configurationFilePath}".`
+            );
+          }
+
+          // Prevent module paths from being specified by limiting the number of '/' characters present
+          if (packageName.match(/^((@[^\/]+\/)|[^@])[^\/]+\/.*$/)) {
+            throw new Error(
+              `Module paths are not supported when using the "packageDir" token ` +
+                (propertyName ? `of property "${propertyName}" ` : '') +
+                `in "${configurationFilePath}". Only a package name is allowed.`
+            );
+          }
+
+          // Resolve to the root of the package (not the module referenced by the package). Substitute
+          // the heft-jest-plugin root if package name matches.
+          const resolvedPackagePath: string =
+            packageName === PLUGIN_PACKAGE_NAME
+              ? OWN_PACKAGE_FOLDER
+              : Import.resolvePackage({
+                  baseFolderPath: path.dirname(configurationFilePath),
+                  packageName
+                });
+          // First entry is the entire match
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(packageDirMatches[0].length));
+          propertyValue = path.resolve(resolvedPackagePath, restOfPath);
+        }
+
+        // Return early, since the remainder of this function is used to resolve module paths
+        if (!options.resolveAsModule) {
+          return propertyValue;
+        }
+
+        // Example:  @rushstack/heft-jest-plugin
+        if (propertyValue === PLUGIN_PACKAGE_NAME) {
+          return OWN_PACKAGE_FOLDER;
+        }
+
+        // Example:  @rushstack/heft-jest-plugin/path/to/file.js
+        if (propertyValue.startsWith(PLUGIN_PACKAGE_NAME)) {
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(PLUGIN_PACKAGE_NAME.length));
+          return path.join(OWN_PACKAGE_FOLDER, restOfPath);
+        }
+
+        return !options.modulePrefix
```

Comment:
`!` `?` `:` is hurting my head here.  The `?` operator is meant to avoid an `if` statement for very simple logic. This ain't simple. :-)